components/provider_ppucd/src/parse_ppucd.rs

```diff
@@ -0,0 +1,450 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE ).
+use std::collections::{HashMap, HashSet};
+use std::iter::Iterator;
+use std::u32;
+
+use crate::support::UnicodeProperties;
+use icu_uniset::provider::UnicodeProperty;
+use icu_uniset::{UnicodeSet, UnicodeSetBuilder};
+
+fn split_line(line: &str) -> Vec<&str> {
+    line.split(';').collect::<Vec<_>>()
+}
```

Comment:
It seems that you just use it as an iterator, so from performance/memory perspective, it would be cleaner if it returned an Iterator, without collecting it into `Vec`.
But I think this is not a runtime code, so optional.